DOC corrected estimator set_params API (#23816)
scikit-learn/scikit-learn: doc/developers/develop.rst

    my_extra_param -> random
    subestimator -> LogisticRegression()

> The ``set_params`` on the other hand takes as input each of the
> arguments of ``__init__``, unpacks them into a dict of the form

Reviewer: Nit: Minor rewording:

```suggestion
On the other hand, ``set_params`` takes the parameters of ``__init__``
as keyword arguments, unpacks them into a dict of the form
```

> ``'parameter': value`` and sets the parameter of the estimator using this dict.

Reviewer 2: ```suggestion
``'parameter': value`` and sets the parameters of the estimator using this dict.
```

> Return value must be an estimator itself.

Reviewer 2: ```suggestion
Return value must be the estimator itself.
```
